src/main/java/io/candydoc/ddd/extract_ddd_concepts/DDDConceptsExtractionService.java

```diff
@@ -64,72 +70,95 @@ private void trackAndApply(List<Event> occurredEvents) {
+      throw new NoBoundedContextNorSharedKernelFound(command.getPackagesToScan());
+    }
+    trackAndApply(boundedContextEvents);
+    trackAndApply(sharedKernelEvents);
+  }
+
+  public void handle(ExtractAggregates command) {
+    log.info("Extract aggregates from {}", command.getPackageToScan());
+    trackAndApply(aggregatesExtractor.extract(command));
   }
 
+  public void handle(ExtractBoundedContexts command) {}
+
   public void handle(ExtractCoreConcepts command) {
     log.info("Extract core concepts from {}", command.getPackageToScan());
     trackAndApply(coreConceptExtractor.extract(command));
   }
 
+  public void handle(ExtractDomainCommands command) {
+    log.info("Extract domain commands from {}", command.getPackageToScan());
+    trackAndApply(domainCommandExtractor.extract(command));
+  }
+
   public void handle(ExtractDomainEvents command) {
     log.info("Extract domain events from {}", command.getPackageToScan());
     trackAndApply(domainEventExtractor.extract(command));
   }
 
-  public void handle(ExtractDomainCommands command) {
-    log.info("Extract domain commands from {}", command.getPackageToScan());
-    trackAndApply(domainCommandExtractor.extract(command));
+  public void handle(ExtractSharedKernels command) {}
+
+  public void handle(ExtractValueObjects command) {
+    log.info("Extract value objects from {}", command.getPackageToScan());
+    trackAndApply(valueObjectExtractor.extract(command));
   }
 
   public void handle(CheckConceptInteractions command) {
     log.info("Check concept interactions from {}", command.getConceptName());
     trackAndApply(interactionChecker.check(command));
   }
 
-  public void handle(ExtractAggregates command) {
-    log.info("Extract aggregates from {}", command.getPackageToScan());
-    trackAndApply(aggregatesExtractor.extract(command));
+  public void apply(AggregateFound event) {
+    this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
   public void apply(BoundedContextFound event) {
+    this.handle(ExtractAggregates.builder().packageToScan(event.getPackageName()).build());
     this.handle(ExtractCoreConcepts.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractDomainEvents.builder().packageToScan(event.getPackageName()).build());
     this.handle(ExtractDomainCommands.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractAggregates.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractDomainEvents.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
+    this.handle(
+        CheckConceptInteractions.builder()
+            .conceptName(event.getPackageName() + ".package-info")
```

Comment:
You should keep the canonical name of the BoundedContext to use it there. The canonical name includes "package-info"